include/nil/crypto3/zk/snark/arithmetization/plonk/constraint_system.hpp

```diff
@@ -83,15 +86,36 @@ namespace nil {
                     plonk_constraint_system(const gates_container_type &gates,
                                             const copy_constraints_container_type &copy_constraints,
                                             const lookup_gates_container_type &lookup_gates = {},
-                                            const lookup_tables_type &lookup_tables = {}
+                                            const lookup_tables_type &lookup_tables = {},
+                                            const public_input_sizes_type public_input_sizes = {}
                     ) :
                         _gates(gates),
                         _copy_constraints(copy_constraints),
                         _lookup_gates(lookup_gates),
-                        _lookup_tables(lookup_tables)
+                        _lookup_tables(lookup_tables),
+                        _public_input_sizes(public_input_sizes)
                     {
                     }
 
+                    std::size_t full_public_input_size() const {
+                        std::size_t sum = 0;
```

Comment:
return  std::accumulate(_public_input_sizes.begin(), _public_input_sizes.end(), 0);

Also maybe rename the function to public_input_total_size ?